front/components/app/PostHogTracker.tsx

```diff
@@ -59,27 +59,29 @@ export function PostHogTracker({ children }: PostHogTrackerProps) {
     };
   }, [activeSubscription]);
 
-  const [hasOptedIn, setHasOptedIn] = useState(false);
-  const lastIdentifiedUserId = useRef<string | null>(null);
-  const lastIdentifiedWorkspaceId = useRef<string | null>(null);
-  const lastPlanPropertiesString = useRef<string | null>(null);
-
-  const isExcludedPath = EXCLUDED_PATHS.some((path) => {
+  const isTrackablePage = !EXCLUDED_PATHS.some((path) => {
     const pathname = router.pathname;
     return pathname.startsWith(path) || pathname.endsWith(path);
   });
-  const isTrackablePage = !isExcludedPath;
 
+  const shouldTrack = isTrackablePage && hasAcceptedCookies;
+
+  const lastIdentifiedUserId = useRef<string | null>(null);
+  const lastIdentifiedWorkspaceId = useRef<string | null>(null);
+  const lastPlanPropertiesString = useRef<string | null>(null);
+  const hasInitialized = useRef(false);
+
+  // Initialize PostHog once with correct default state
   useEffect(() => {
-    if (!POSTHOG_KEY || posthog.__loaded) {
+    if (!POSTHOG_KEY || posthog.__loaded || hasInitialized.current) {
```

Comment:
nit but it might be a bit easier to understand if you only have one useEffect instead of having two since they have the exactly same dependency (one for init and one for updating)?

Reply:
but maybe it's too long and actually more complicated 🙃